packages/discovery-provider/src/tasks/index_payment_router.py

```diff
@@ -0,0 +1,859 @@
+import concurrent.futures
```

Comment:
Might be worth comparing this file to `index_user_bank.py` if you're not familiar with the indexing. A lot of this logic was copied and modified from there. It could use cleanup later on and we could potentially move all of the logic in the second half of the file around fetching and caching transactions/slots into a shared wrapper of some sort.

Reply:
there have been many talks about templatizing indexers...